docs/developer-guide/implement-a-udf.rst

```diff
@@ -0,0 +1,349 @@
+.. _implement-a-udf:
+
+Implement a User-defined Function (UDF)
+#######################################
+
+Prerequisites
+     - `Apache Maven <https://maven.apache.org/download.cgi>`__
+     - |cp| installed locally
+     - Internet connectivity for downloading Confluent POM files
+
+Create the KSQL Extensions Directory
+************************************
+
+When you create a custom user-defined function (UDF), you implement it in Java
+and deploy it as a JAR to the KSQL extensions directory. By default, this 
+directory doesn't exist, so you need to create it and assign it in the KSQL
+Server configuration properties.
+
+Create the KSQL extensions directory, ``<path-to-confluent>/etc/ksql/ext``:
+
+.. codewithvars:: bash
+
+    mkdir confluent-|release|/etc/ksql/ext
+
+Edit the ``ksql-server.properties`` configuration file in
+``<path-to-confluent>/etc/ksql`` to add the fully qualified path to the
+``ext`` directory:
+
+.. codewithvars:: text
+
+    ksql.extension.dir=/home/my-home-dir/confluent-|release|/etc/ksql/ext
+
+.. note::
+
+    Use the fully qualified path. KSQL Server won't load extensions if the 
+    path begins with ``~``.
+
+
+Create the Source and Project Files
+***********************************
+
+The following steps shows how to implement your UDF in a Java class and build
+it by defining a Maven POM file.
+
+#. Create a root directory for your UDF's source code and project files.

```

Comment:
i was expecting theses steps to mimic the headings that follow. shouldn't they? and if so, consider making them hyperlinks to the actual headings.